src/app/main/component/events/components/event-details/event-details.component.ts

```diff
@@ -122,6 +128,36 @@ export class EventDetailsComponent implements OnInit, OnDestroy {
     }
   }
 
+  private getIsLiked(): void {
+    this.eventService
+      .getIsLikedByUser(this.eventId)
+      .pipe(take(1))
+      .subscribe((isLiked: boolean) => {
+        this.isLiked = isLiked;
+      });
+  }
+
+  onLikeEvent(): void {
+    const updatedLikes = this.isLiked ? this.event.likes - 1 : this.event.likes + 1;
```

Comment:
can refactor it to a simpler approach:
`  const updatedLikes = this.event.likes + (this.isLiked ? 1 : -1);`